core/src/test/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoaderTest.java

```diff
@@ -0,0 +1,104 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.os.Handler;
+import com.google.android.exoplayer2.source.ads.AdsLoader;
+import com.novoda.noplayer.AdvertBreak;
+import com.novoda.noplayer.AdvertsLoader;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.ArgumentCaptor;
+import utils.ExceptionMatcher;
+
+import java.util.Arrays;
+
+import static com.novoda.noplayer.AdvertBreakFixtures.anAdvertBreak;
+import static com.novoda.noplayer.AdvertFixtures.anAdvert;
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class NoPlayerAdsLoaderTest {
```

Comment:
I guess we can't really test anything using the `exo-player` listeners? Or do you think that goes too much into implementation details without going for the behaviour that we care about?

Reply:
I tried and it seemed a bit forced because it would look like this:
call `adsLoader.bind(advertListener)`
call `adsLoade.start()`
capture `advertsLoadedCallback`
call `advertsLoadedCallback.onAdvertsLoaded()` with correct adverts
setup the `player` so it returns reasonable values

call `adsLoader.onPositionDiscontinuity()` or `adsLoader.onTimelineChanged()` with correct parameters 

assert that correct `advertListener` method was called

😅

Reply:
Yea I figured it might look a bit weird 😂 fair enough.